react-intl/react-intl-tests.tsx

```diff
@@ -60,6 +56,17 @@ const SomeFunctionalComponentWithIntl = injectIntl<SomeComponentProps>(({
     );
 });
 
+@injectIntl
```

Comment:
For now that is the best you can do with typescript as of version 2.2. Not only this library has incomplete signature for decorator function. Take a look at `react-redux`, `redux-form` for example.